scripts/repo-variable-writer.go

```diff
@@ -0,0 +1,373 @@
+	}
+
+	return nil
+}
+
+func (obj *GithubVariable) ReadRepoVariable(projName string) (*string, error) {
+
+	variableName := obj.genVariableName(projName)
+	url, header, err := obj.genUrlAndHeaders(ghRepoVariableEndpoint, variableName, "")
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := handleHTTPCall(
+		http.MethodGet,
+		*url,
+		time.Minute,
+		nil,
+		header,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("status code was not 200, got: %v", resp.StatusCode)
+	}
+
+	var variableData struct {
+		VariableValue string `json:"value"`
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&variableData); err != nil {
+		return nil, fmt.Errorf("failed deserialize response body: %v", err)
+	}
+
+	return toPtr[string](variableData.VariableValue), nil
+}
+
+func (obj *GithubVariable) UpdatedProjectDispatcher(projData Project, version string) error {
+
+	projToWorkflowTable := map[string]string{
+		"falco": "falco.yaml",
+	}
+
+	workflowFileName, found := projToWorkflowTable[projData.Name]
+	if !found {
+		return fmt.Errorf("failed to find the workflow file related to project: %s", projData.Name+"/"+projData.Organization)
+	}
+
+	url, header, err := obj.genUrlAndHeaders(ghWorkflowDispatchEndpoint, "", workflowFileName)
+	if err != nil {
+		return err
+	}
+
+	type WorkflowData struct {
+		Ref    string `json:"ref"`
+		Inputs struct {
+			CncfProject    string `json:"cncf_project"`
+			Version        string `json:"version"`
+			CncfProjectSub string `json:"cncf_project_sub"`
+		} `json:"inputs"`
+	}
+
+	for _, subComponent := range projData.SubComponents {
+		workflowData := WorkflowData{}
+		workflowData.Inputs.CncfProject = projData.Name
+		workflowData.Ref = "main" // TODO: need to figure it out
+		workflowData.Inputs.Version = version
+		workflowData.Inputs.CncfProjectSub = subComponent
+
+		var _newWorkflowData bytes.Buffer
+
+		if err := json.NewEncoder(&_newWorkflowData).Encode(workflowData); err != nil {
+			return fmt.Errorf("failed to serialize the body: %v", err)
+		}
+
+		resp, err := handleHTTPCall(
+			http.MethodPost,
+			*url,
+			time.Minute,
+			&_newWorkflowData,
+			header,
+		)
+		if err != nil {
+			return err
+		}
+
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusNoContent {
+			return fmt.Errorf(
+				"failed in workflowDispatch, status code was not 204, got: %v",
+				resp.StatusCode)
+		}
+
+		slog.Info("Successfully workflow_dispatch",
+			"Proj", projData.Name,
+			"Org", projData.Organization,
+			"Ver", version,
+			"Subcomponent", subComponent)
+	}
+
+	return nil
+}
+
+func main() {
+
+	projects, err := loadProjectMetadata(LocProjMeta)
+	if err != nil {
+		slog.Error("Failed: unable to read the projects Metadata", "Reason", err)
+		os.Exit(1)
+	}
+	slog.Info("Got projects metadata", "Projects", projects)
+
+	storage, err := NewGithubVariable()
```

Comment:
storage?

Reply:
I'm considering github repo variables as a persistent storage, thats why this name